_posts/2024-12-19-moving-data-between-JBOD-disks-using-cruise-control.md

```diff
@@ -0,0 +1,345 @@
+    excludedBrokersForReplicaMove: []
+    excludedTopics: []
+    intraBrokerDataToMoveMB: 0
+    monitoredPartitionsPercentage: 100
+    numIntraBrokerReplicaMovements: 26
+    numLeaderMovements: 0
+    numReplicaMovements: 0
+    onDemandBalancednessScoreAfter: 100
+    onDemandBalancednessScoreBefore: 0
+    provisionRecommendation: ""
+    provisionStatus: UNDECIDED
+    recentWindows: 1
+  sessionId: 24537b9c-a315-4715-8e86-01481e914771
+```
+
+Now you can use the `approve` annotation to apply the generated proposal.
+Once the rebalance is complete, you can then use the `kafka-log-dirs.sh` tool again to check if the data moved or not:
+```shell
+ kubectl exec -n myproject -ti my-cluster-pool-a-0   /bin/bash -- bin/kafka-log-dirs.sh --describe --bootstrap-server my-cluster-kafka-bootstrap:9092  --broker-list 0 --topic-list my-topic
+ ```
+
+Output:
+```yaml
+{
+  "brokers": [
+    {
+      "broker": 0,
+      "logDirs": [
+        {
+          "partitions": [],
+          "error": null,
+          "logDir": "/var/lib/kafka/data-2/kafka-log0"
+        },
+        {
+          "partitions": [
+            {
+              "partition": "my-topic-4",
+              "size": 0,
+              "offsetLag": 0,
+              "isFuture": false
+            },
+            {
+              "partition": "my-topic-5",
+              "size": 0,
+              "offsetLag": 0,
+              "isFuture": false
+            },
+            {
+              "partition": "my-topic-0",
+              "size": 0,
+              "offsetLag": 0,
+              "isFuture": false
+            },
+            {
+              "partition": "my-topic-1",
+              "size": 0,
+              "offsetLag": 0,
+              "isFuture": false
+            },
+            {
+              "partition": "my-topic-2",
+              "size": 0,
+              "offsetLag": 0,
+              "isFuture": false
+            },
+            {
+              "partition": "my-topic-3",
+              "size": 0,
+              "offsetLag": 0,
+              "isFuture": false
+            }
+          ],
+          "error": null,
+          "logDir": "/var/lib/kafka/data-0/kafka-log0"
+        },
+        {
+          "partitions": [],
+          "error": null,
+          "logDir": "/var/lib/kafka/data-1/kafka-log0"
+        }
+      ]
+```
+
+As you can see in the output, the data has moved from volume 1 and volume 2, and they now don’t have any partition replicas on them. All the partition replicas are now moved to volume 0 on broker with ID 0.
+
+### Additional notes
+
+1. This feature only works if JBOD storage is enabled
+2. Make sure you have more than one volume per broker else you will be prompted of not having enough volumes to move the data to.
+3. This endpoint does not provide `before` load since upstream Cruise Control project does not support `verbose` with this endpoint so the `loadmap` generated should only have `afterLoad` information.
+
+## What's next
```

Comment:
You should probably list the risks / incomplete parts as well ... in paritcular, new partition replicas might be scheduled to the disks between cleaning them up with cruise control and removing them which might lead to data loss again.